shared/Microsoft.Extensions.TypeNameHelper.Sources/TypeNameHelper.cs

```diff
@@ -115,9 +117,15 @@ private static void ProcessGenericType(StringBuilder builder, Type type, Type[]
             for (var i = offset; i < length; i++)
             {
                 ProcessType(builder, genericArguments[i], fullName);
-                if (i + 1 != length)
+                if (i + 1 == length)
+                {
+                    continue;
+                }
+
+                builder.Append(',');
+                if (!genericArguments[i + 1].IsGenericParameter)
```

Comment:
If one parameter is generic then all of them are generic. Check for one and just do string join for rest and skip looping fully.

Reply:
There are cases when one parameter open and another close, for example:
```
class TestType<T> : BaseType<int,T>
{
}
```

Reply:
typeof(TestType<>) has only 1 generic parameter.

Reply:
If it is possible to create, it is good logic, but we should also add test for it.

Reply:
`typeof(TestType<T>).BaseType` will have one open and one close parameter (`int` and `T`), unit test [here](https://github.com/aspnet/Common/pull/283/files#diff-65ffadf8398a22a547ef1fc2cf411cb5R152) done to test this logic.